File: langchain-server/types/request.go
package types

type RequestChatBody struct {
	Credential   RequestChatCredential `json:"credential"`
	SystemPrompt *string               `json:"system_prompt,omitempty"`
	Input        string                `json:"input"`
	Servers      []string              `json:"servers"`
}

type RequestChatCredential struct {
	Provider string  `json:"provider"`
	Model    *string `json:"model,omitempty"`
	URL      *string `json:"url,omitempty"`
	APIKey   *string `json:"api_key,omitempty"`
	Set      *SetLLM `json:"set,omitempty"`
}

type SetLLM struct {
	Temperature        *float64 `json:"temperature,omitempty"`
	MaxTokens          *int     `json:"max_tokens,omitempty"`
	TopP               *float64 `json:"top_p,omitempty"`
	FrequencyPenalty   *float64 `json:"frequency_penalty,omitempty"`
	PresencePenalty    *float64 `json:"presence_penalty,omitempty"`
	Stop               []string `json:"stop,omitempty"`
	Timeout            *int     `json:"timeout,omitempty"`
	MaxRetries         *int     `json:"max_retries,omitempty"`
	MaxContextMessages *int     `json:"max_context_messages,omitempty"` // Limit message history to prevent context overflow
}

type LLMPublicProvider string

const (
	ProviderOpenAI     LLMPublicProvider = "openai"
	ProviderClaude     LLMPublicProvider = "claude"
	ProviderOpenRouter LLMPublicProvider = "openrouter"
)

type LLMLocalProvider string

const (
	ProviderOllama   LLMLocalProvider = "ollama"
	ProviderLlamaCPP LLMLocalProvider = "llama_cpp"
	ProviderVLLM     LLMLocalProvider = "vllm"
)

File: types/request.ts
export interface RequestChatBody {
  credential: RequestChatCredential;
  system_prompt?: string;
  input: string;
  servers: string[];
}

export interface RequestChatCredential {
  provider: LlmPublicProvider | LlmLocalProvider;
  model?: string; // jika tidak di isi maka masing2 punya default
  url?: string;
  api_key?: string;
  set?: SetLlm;
}

export interface SetLlm {
  temperature?: number; // 0.0 - 2.0, default: 0
  max_tokens?: number; // Maximum tokens to generate
  top_p?: number; // 0.0 - 1.0, nucleus sampling
  frequency_penalty?: number; // -2.0 - 2.0, penalize frequent tokens
  presence_penalty?: number; // -2.0 - 2.0, penalize repeated tokens
  stop?: string[]; // Stop sequences
  timeout?: number; // Request timeout in ms
  max_retries?: number; // Number of retries on failure
}

export type LlmPublicProvider = "openai" | "claude" | "openrouter";
export type LlmLocalProvider = "ollama" | "llama_cpp" | "vllm";
